src/hmem_rocr.c

```diff
@@ -65,18 +65,27 @@ struct rocr_ops {
 	hsa_status_t (*hsa_agent_get_info)(hsa_agent_t agent,
 					   hsa_agent_info_t attribute,
 					   void *value);
+	hsa_status_t (*hsa_amd_ipc_memory_create)(void* ptr, size_t len,
+					hsa_amd_ipc_memory_t* handle);
+	hsa_status_t (*hsa_amd_ipc_memory_attach)(
+		const hsa_amd_ipc_memory_t* handle, size_t len,
+		uint32_t num_agents,
+		const hsa_agent_t* mapping_agents,
+		void** mapped_ptr);
+	hsa_status_t (*hsa_amd_ipc_memory_detach)(void* mapped_ptr);
 };
 
 #if ENABLE_ROCR_DLOPEN
 
 #include <dlfcn.h>
 
-static void *rocr_handle;
-static struct rocr_ops rocr_ops;
+static void *hsa_handle;
+static struct hsa_ops hsa_ops;
```

Comment:
Please move separate updates into separate patches to make it easier to review and cherry pick changes